### DISCUSSION TITLE
Graphical artifacts with fractional scaling on Linux

### DISCUSSION THREAD

**Discussion by lbrame** [ghostty-org/ghostty#3478] [Issue Triage] [answered]
I am using Ghostty built on Linux on a Fedora 41 desktop running GNOME 47, Wayland session, and the scaling level set to 175%. My build is compiled with the optimizer on from commit [a8e5eef](https://github.com/ghostty-org/ghostty/commit/a8e5eef11cc67f87f445626f9ca2993373774bf8). On top of the already known issue #1938 , I have also used numerous artifacts that haappen during regular use. An example is random white or transparent lines everywhere:

![Image](https://github.com/user-attachments/assets/f623b568-5b5a-4df2-9263-9a93378933f1)

![Image](https://github.com/user-attachments/assets/6d90af1c-f874-4933-a5b7-2568ff018270)

Another issue I have seen is that, when the terminal is tiled to an edge of the screen on GNOME, it does not fully adhere to the panel and neighbor windows, but it has a visible gap.

![Image](https://github.com/user-attachments/assets/fe096902-98dd-4367-b266-a617732a14c5)

This only happens on the GTK build. The basic `glfw` output produced by `zig build -Dapp-runtime=glfw run` is unaffected:

![Image](https://github.com/user-attachments/assets/4b09280b-cc47-42d2-a8c0-dd0559902abb)

**Comment by mitchellh** [comment]
Converted to a discussion to figure out what a good path forward is for this. This may also be considered a dupe because resolving this may also resolve fractional issues for fonts too.

**Comment by shuhaowu** [comment]
I have the same issue as well. It seems to mostly affect the UI chrome from my experiments and not the actual terminal part as the OP mentioned.

https://github.com/user-attachments/assets/4e2a54c0-3a5b-4f77-8c9e-80b090a1eef9

**Comment by momoRaza** [comment]
I'm also having the same issue on latest Arch Linux, Gnome 47 with fractional scaling 
![Screenshot From 2025-01-02 13-00-12](https://github.com/user-attachments/assets/519fa1ae-439f-40e4-939d-9fbc3ecb01cf)
![Screenshot From 2025-01-02 12-59-57](https://github.com/user-attachments/assets/6bb5cad0-8523-4fb6-9679-9f05c9d716c8)

**Comment by JannesStroehlein** [comment]
Hi,
I also have the issue on Arch with KDE using Wayland. At least for me this issue seems to only occur in GTK apps. 
When using 100% scaling no artifacts occur, but at 130% I also get artifacts.

**Comment by DarkArc** [comment]
Yeah, on KDE Wayland (Fedora) with 1.5x scale factor I see this when I run the mouse over various buttons, there's often artifacting:

![image](https://github.com/user-attachments/assets/37b261dc-8a23-450f-9551-d6032249c328)

I'm not noticing any issues with font rendering, just the GTK portions of the UI (+ the oversized cursor issue).

**Comment by pluiedev** [comment, marked as answer]
Looks like a GTK artifact (especially since it doesn't affect the actual terminal itself)

**Comment by ku1ik** [comment]
Data point: I had similar artifacts on Gnome/Wayland/fractional scaling with ghostty 1.0, but with the font sharpness fix that came with 1.0.1 or 1.0.2 (I don't remember exactly) the problem went away for me.

**Comment by 00-kat** [comment]
For the record, I've seen weird graphical artifacts in the right click menu and context menu in **every** GTK4 + libadwaita app, and not just Ghostty, when using fractional scaling.

**Comment by 00-kat** [comment]
Cross-referencing #3466.
